nbqa/config_parser.py

```diff
@@ -113,16 +120,21 @@ def parse_config_from_file(cli_args: CLIArgs, project_root: Path) -> Optional[Co
     Optional[Configs]
         Configuration read from the file(if any)
     """
-    config: Optional[Configs] = None
-
     for config_file, config_handler in _CONFIG_FILE_HANDLERS:
         file_path: Path = project_root / config_file
 
         if file_path.is_file() and config_handler is not None:
             config = config_handler(cli_args.command, file_path)
 
-        # we found the config. skip other files
-        if config is not None:
-            break
+            # we found the config. skip other files
+            if config is not None:
+                break
+    else:
+        config = Configs()
+
+    for section in CONFIG_SECTIONS:
+        default = DEFAULT_CONFIG[section].get(cli_args.command, None)
+        if not getattr(config, section, None):
```

Comment:
My understanding of this change is to always use `treat comment as code` option for isort. But in the above change if the user is passing some options to `isort` then `treat comment as code` option won't be set. I thought `treat comment as code` option would be appended to options passed by user. Basically `user_config.merge(default_config)`

Reply:
oops, you're right, I'll just append it, thanks - I need a new test for this case then :)

Reply:
@girip11 I was thinking about this again, and am _tempted_ to say that instead we should have some default configs for each tools which the user can then _overwrite_ (this'll be made explicit in the documentation - and if anyone will use pyproject.toml for configuring nbqa, it means they're already reading the docs :) ).

This way, for each tool, we can provide what we think are some sensible configs (similar to the ones we have in the examples from the readme), and the user can overwrite them if they wish. There may indeed be a reason someone doesn't want to use e.g. `--treat-comment-as-code` or any of the other default configs we will provide

Reply:
If we are making the configuration as optional (that can be overridden by the user), then we really don't have to enforce isort version right? Since the PR title says `always pass treat comment as code`, I thought this change is adding a few configurations always on top of user provided configuration. Now we are at `Forcing some defaults vs Providing some sensible defaults`. If you are inclining towards providing sensible defaults rather than forcing those, then we should not raise exception when isort version is not above '5.3.0'. Rather we should do something like add these default configuration only when the tools version is more than so and so otherwise still run the tool. Could you elaborate more on this at issue #163 itself.

Reply:
:thinking: I should've thought this through before starting...I can't see why anyone would _not_ want `--treat-comment-as-code`, so perhaps we can treat this PR as "adding a few configurations always on top of user provided configuration". We can start with this `isort` one, and then perhaps extend it to others later (e.g. always ignore the 'trailing semicolon' error from flake8)